File: BooksTest/app/src/main/java/com/rshtukaraxondevgroup/bookstest/database/AppDatabase.java
package com.rshtukaraxondevgroup.bookstest.database;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;

import com.rshtukaraxondevgroup.bookstest.Constants;
import com.rshtukaraxondevgroup.bookstest.model.BookModel;

@Database(entities = {BookModel.class}, version = 1)
public abstract class AppDatabase extends RoomDatabase {
    public abstract BookDao bookDao();

    private static AppDatabase INSTANCE;

    public static AppDatabase getDatabase(final Context context) {
        if (INSTANCE == null) {
            synchronized (AppDatabase.class) {
                if (INSTANCE == null) {
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                            AppDatabase.class, Constants.DATABASE)
                            .build();

                }
            }
        }
        return INSTANCE;
    }
}

File: RandomUsersTest/app/src/main/java/com/skaravatskiy/randomusers/data/model/database/AppDataBase.kt
package com.skaravatskiy.randomusers.data.model.database

import android.arch.persistence.room.Database
import android.arch.persistence.room.RoomDatabase


@Database(entities = [User::class], version = 1)
abstract class AppDatabase : RoomDatabase() {
    abstract fun daoAccess(): UsersDao
}

